### DISCUSSION TITLE
[RFC] Build and install using cmake and meson native commands

### DISCUSSION THREAD

**Discussion by yurikoles** [Homebrew/discussions#3327] [Tap maintenance and Homebrew development] [answered]
Hi,

The new template for `install` created with `brew create --cmake` looks close to upstream recommendations, i.e. like this
```ruby
def install
  # ENV.deparallelize  # if your formula fails when building in parallel
  system "cmake", "-S", ".", "-B", "build", *std_cmake_args
  system "cmake", "--build", "build"
  system "cmake", "--install", "build"
end
```
I'm volunteering to port existing formulae to this pattern.

For `brew create --meson` I see:
```ruby
def install
  # ENV.deparallelize  # if your formula fails when building in parallel
  mkdir "build" do
    system "meson", *std_meson_args, ".."
    system "ninja", "-v"
    system "ninja", "install", "-v"
  end
end
```
I'm volunteering to adjust template to use native `meson` commands and to port existing formulae to this patter, e.g.:
```ruby
def install
  # ENV.deparallelize  # if your formula fails when building in parallel
  system "meson", ".", "build", *std_meson_args
  system "meson", "compile", "-C", "build"
  system "meson", "install", "-C", "build"
end
```

After that ports will be finished, it would be nice to have some `rubocop` rules to warn maintainers about using deprecate patterns, or even add some logic that will automatically modernize them.

**Comment by carlocab** [comment, marked as answer]
I'm open to making the changes for formulae that use `meson`. I have been contemplating them for some time now.

However, we probably want to do this first:

> it would be nice to have some `rubocop` rules to warn maintainers about using deprecate patterns, or even add some logic that will automatically modernize them.

Arguably this should've been done for `cmake` first too, but that ship has sailed.
